tests/unit/test_irc_bridge.py

```diff
@@ -1,354 +1,128 @@
-    harness.charm._irc_bridge._on_collect_status(event_mock)
+from charm_state import CharmState, SynapseConfig
+from charm_types import DatasourcePostgreSQL
+from irc_bridge import enable_irc_bridge
+from synapse import ExecResult
 
-    event_mock.add_status.assert_not_called()
-    enable_irc_bridge_mock.assert_not_called()
 
+@pytest.fixture(name="state")
+def charm_state_fixture():
+    """Construct a CharmState object."""
 
-def test_on_collect_status_no_service(harness: Harness, monkeypatch: pytest.MonkeyPatch) -> None:
-    """
-    arrange: start the Synapse charm, set server_name, mock get_services to return a empty dict.
-    act: call _on_collect_status.
-    assert: enable_irc_bridge is called and status is set to active as
-    the bridge can start before Synapse.
-    """
-    harness.update_config({"enable_irc_bridge": True})
-    harness.begin_with_initial_hooks()
-    harness.set_leader(True)
-    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
-    monkeypatch.setattr(container, "get_services", MagicMock(return_value={}))
-    enable_irc_bridge_mock = MagicMock()
-    enable_irc_bridge_mock.enable_irc_bridge = True
-    monkeypatch.setattr(IRCBridgeObserver, "_enable_irc_bridge", enable_irc_bridge_mock)
+    def charm_state_with_db(with_db_config: bool) -> CharmState:
+        """
+        Create a CharmState object with a SynapseConfig and optional DatasourcePostgreSQL object.
 
-    event_mock = MagicMock()
-    harness.charm._irc_bridge._on_collect_status(event_mock)
-
-    assert isinstance(harness.model.unit.status, ops.ActiveStatus)
-    enable_irc_bridge_mock.assert_called_once()
-
-
-def test_on_collect_status_container_off(
-    harness: Harness, monkeypatch: pytest.MonkeyPatch
-) -> None:
-    """
-    arrange: start the Synapse charm, set server_name, mock container to not connect.
-    act: call _on_collect_status.
-    assert: no actions is taken because the container is off.
-    """
-    harness.update_config({"enable_irc_bridge": True})
-    harness.begin_with_initial_hooks()
-    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
-    monkeypatch.setattr(container, "can_connect", MagicMock(return_value=False))
-    enable_irc_bridge_mock = MagicMock()
-    monkeypatch.setattr(IRCBridgeObserver, "_enable_irc_bridge", enable_irc_bridge_mock)
-
-    event_mock = MagicMock()
-    harness.charm._irc_bridge._on_collect_status(event_mock)
-
-    event_mock.add_status.assert_not_called()
-    enable_irc_bridge_mock.assert_not_called()
+        Args:
+            with_db_config: Whether to include a DatasourcePostgreSQL object in the CharmState.
 
+        Returns:
+            A CharmState object.
+        """
+        synapse_config = SynapseConfig(  # type: ignore
+            server_name="foo",
+        )
+        db_config = DatasourcePostgreSQL(
+            user="bar",
+            password="baz",  # nosec
```

Comment:
we should use token_hex (same for all other cases https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#random-values